From 109531ca0e7c247082019292c5c27f38498e5acb Mon Sep 17 00:00:00 2001
From: Ilya Konstantinov <ilya.konstantinov@gmail.com>
Date: Thu, 7 May 2015 01:16:36 +0200
Subject: [PATCH 5/5] interfaces: inhibit IPv6 unless
 LIBNICE_GATHER_IPV6_CANDIDATES

For current practical purposes, it only:
- wastes time
- makes the candidate lists longer and more complicated
- risks additional bugs related to IPv6
---
 agent/interfaces.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/agent/interfaces.c b/agent/interfaces.c
index 0fa2fd7..62a5568 100644
--- a/agent/interfaces.c
+++ b/agent/interfaces.c
@@ -120,7 +120,11 @@ nice_interfaces_get_local_interfaces (void)
     if (ifa->ifa_addr == NULL)
       continue;
 
+#ifndef LIBNICE_GATHER_IPV6_CANDIDATES
+    if (ifa->ifa_addr->sa_family == AF_INET) {
+#else
     if (ifa->ifa_addr->sa_family == AF_INET || ifa->ifa_addr->sa_family == AF_INET6) {
+#endif
       nice_debug ("Found interface : %s", ifa->ifa_name);
       interfaces = g_list_prepend (interfaces, g_strdup (ifa->ifa_name));
     }
@@ -259,6 +263,11 @@ nice_interfaces_get_local_ips (gboolean include_loopback)
     if (ifa->ifa_addr == NULL)
       continue;
 
+#ifndef LIBNICE_GATHER_IPV6_CANDIDATES
+    if (ifa->ifa_addr->sa_family != AF_INET)
+      continue;
+#endif
+
     /* Convert to a string. */
     addr_string = sockaddr_to_string (ifa->ifa_addr);
     if (addr_string == NULL) {
-- 
2.2.1

